### ISSUE TITLE
array_keys() issue only with phar version

### ISSUE THREAD

**Issue by Rarst** [vimeo/psalm#2858]
For this code (a bit too much going on to reproduce in sandbox):

```php
	/**
	 * @return Stopwatch_Record[]
	 */
	public function get_records(): array {

		/** @psalm-var array<string, StopwatchEvent> $events */
		$events = $this->stopwatch->getSectionEvents( '__root__' );

		/** @var string $name */
		foreach ( array_keys( $events ) as $name ) {
			if ( $this->stopwatch->isStarted( $name ) ) {
				unset( $events[ $name ] );
			}
		}

		return array_map( [ $this, 'transform' ], array_keys( $events ), $events );
	}

	/**
	 * @param string         $name  Event name.
	 * @param StopwatchEvent $event Stopwatch event instance.
	 *
	 * @return Stopwatch_Record
	 */
	protected function transform( string $name, StopwatchEvent $event ): Stopwatch_Record {

		return new Stopwatch_Record( $name, $event );
	}
```
https://github.com/Rarst/laps/blob/168a226859dade7b431d2d69ba751fb3667ee146/src/Record/Collector/Stopwatch_Collector.php#L49-L76

Getting this error:

> ERROR: InvalidScalarArgument - src\Record\Collector\Stopwatch_Collector.php:64:21 - First parameter of closure passed to function array_map expects string, int|string provided
                return array_map( [ $this, 'transform' ], array_keys( $events ), $events );

However `array_keys( $events )` should only contain/pass strings since `$events` is documented to only have string keys above `array<string, StopwatchEvent>`.

**Comment by Rarst** [comment]
Hmm, seems to not happen on master, but I am not finding recent issues about `array_keys()`...

**Comment by Rarst** [comment]
Okay, this got stranger — I can reproduce this, but _only_ with phar version, even if it has exact same commit version as git checkout. Here is running both on current master of the repo (`psalm` is phar, `psalm-dev` is git checkout):

```
c:\server\www\dev\wp-content\plugins\laps>psalm --version & psalm-dev --version &&  psalm & psalm-dev
Psalm 3.9.5@0cfe565d0afbcd31eadcc281b9017b5692911661
Psalm 3.x-dev@0cfe565d0afbcd31eadcc281b9017b5692911661
Scanning files...
Analyzing files...

░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��EE░��░��░��░��░��

ERROR: InvalidScalarArgument - src\Record\Collector\Sql_Collector.php:68:39 - First parameter of closure passed to function array_map expects int, int|string provided
                $records = array_filter( array_map( [ $this, 'transform' ], array_keys( $wpdb->queries ), $wpdb->queries ) );


ERROR: InvalidScalarArgument - src\Record\Collector\Stopwatch_Collector.php:64:21 - First parameter of closure passed to function array_map expects string, int|string provided
                return array_map( [ $this, 'transform' ], array_keys( $events ),
$events );


------------------------------
2 errors found
------------------------------

Checks took 4.41 seconds and used 56.625MB of memory
Psalm was able to infer types for 95.1049% of the codebase
Scanning files...
Analyzing files...

░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��░��

------------------------------
No errors found!
------------------------------

Checks took 4.69 seconds and used 62.938MB of memory
Psalm was able to infer types for 96.1583% of the codebase
```

**Comment by Rarst** [comment]
Possibly related #2953 since on Windows git defaults to checkout with Windows line endings (but commit with Unix line endings).

So phar copy would have original Unix line endings, but git copy would have them changed to Windows ones.

**Comment by Rarst** [comment]
I tried replacing line endings to Unix ones in source PHP files, but still get the issue. :\

**Comment by veewee** [comment]
I've created a POC to reproducte the error:
https://github.com/veewee/poc-psalm/actions/runs/172331087

It happens while using the phar version on windows.
For mac + linux it works fine.

**Comment by weirdan** [comment]
I suspect it's fixed in #10748. @veewee can you try with this phar: https://output.circle-artifacts.com/output/job/42a4e96a-0be0-4c25-a2f0-8eb99a79ecd8/artifacts/0/build/psalm.phar

PS: yes, it took me 4 years to reboot to Windows.
